case_utils/case_validate/__init__.py

```diff
@@ -32,141 +32,90 @@
 __version__ = "0.3.0"
 
 import argparse
-import importlib.resources
 import logging
 import os
 import sys
 import warnings
-from typing import Dict, Set, Tuple, Union
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 import pyshacl  # type: ignore
 import rdflib
+from rdflib import Graph
 
-import case_utils.ontology
+from case_utils.case_validate.validate_types import (
+    NonExistentCDOConceptWarning,
+    ValidationResult,
+)
+from case_utils.case_validate.validate_utils import (
+    get_invalid_cdo_concepts,
+    get_ontology_graph,
+)
 from case_utils.ontology.version_info import (
     CURRENT_CASE_VERSION,
     built_version_choices_list,
 )
 
-NS_OWL = rdflib.OWL
-NS_RDF = rdflib.RDF
-NS_RDFS = rdflib.RDFS
-NS_SH = rdflib.SH
-
 _logger = logging.getLogger(os.path.basename(__file__))
 
 
-class NonExistentCDOConceptWarning(UserWarning):
+def validate(
+    input_file: str,
```

Comment:
For our awareness, this is a narrower argument than the first argument of `pyshacl.validate`; here's today's definition:

https://github.com/RDFLib/pySHACL/blob/v0.23.0/pyshacl/validate.py#L369-L370

`pyshacl.validate`'s first argument seems to permit a string to be a file path or URL, OR a full string dump of a graph.  See these lines for heuristics in `pyshacl.rdfutil.load.load_from_graph`:

https://github.com/RDFLib/pySHACL/blob/v0.23.0/pyshacl/rdfutil/load.py#L222-L227

Should we implement "`str` means path" now, or just adopt the `load_from_graph` usage now from these lines:

https://github.com/RDFLib/pySHACL/blob/v0.23.0/pyshacl/validate.py#L424-L428

Reply:
Welp, I found an issue pushing us towards expanding `input_file` from `str`.  `case_validate` is written to take multiple input files as data graphs (as well as multiple input files as ontology graphs).  I believe this behavior should be preserved, because otherwise a user that needs to read two data graphs at once needs to do some intermediary graph compilation before calling `case_validate`.

So, I think the first argument needs to become at least either `Union[str, List[str]]` or `Union[str, Graph]`.  The current code path from the CLI entry point I think favors `Union[str, List[str]]`.

Reply:
I believe this is addressed in `15f00c9`.